server/src/engine/PhasesManager/PhasesManager.ts

```diff
@@ -0,0 +1,37 @@
+import { Phases } from "@global/Phases";
+import Game from "../Game";
+import { phaseHandlers } from "./handlers";
+
+export type PhaseHandler = {
+  duration: number | null; // Duration of the phase in milliseconds or null if no timeout
+  onEnter(game: Game): void; // Called when the phase starts
+  transition(game: Game, isTimeup: boolean): Phases | null; // Return the next phase or null if the phase should not change
+};
+
+export class PhasesManager {
+  private _phase: Phases = Phases.LOBBY; // Current phase
+  private phaseHandlers: Record<Phases, PhaseHandler> = phaseHandlers;
+
+  // Extra informations
+  private _day: number = 0;
+  private _night: number = 0;
+
+  constructor() {}
+
+  // Phase management
+  get current() {
+    return this._phase;
+  }
+
+  update(game: Game): void {
+    const nextPhase = this.phaseHandlers[this._phase].transition(game, !game._timer.isRunning);
```

Comment:
```suggestion
    const nextPhase = this.phaseHandlers[this._phase].transition(game, !game._timer.running);
```

See Timer.ts for more context